src/test/java/org/apache/commons/collections4/list/CopyOnWriteListTest.java

```diff
@@ -0,0 +1,417 @@
+            integerList.add(param, 11);
+        }
+    };
+
+    private class AddAllElementTask implements Task<Integer, List<Integer>> {
+        @Override
+        public void execute(Integer param, List<Integer> integerList) {
+            integerList.addAll(Arrays.asList(14, 15));
+        }
+    };
+
+    private class InsertAllElementTask implements Task<Integer, List<Integer>> {
+        @Override
+        public void execute(Integer param, List<Integer> integerList) {
+            integerList.addAll(param, Arrays.asList(14, 15));
+        }
+    };
+
+    private class RemoveIndexedElementTask implements Task<Integer, List<Integer>> {
+        @Override
+        public void execute(Integer param, List<Integer> integerList) {
+            integerList.remove(param.intValue());
+        }
+    };
+
+    private class RemoveObjElementTask implements Task<Integer, List<Integer>> {
+        @Override
+        public void execute(Integer param, List<Integer> integerList) {
+            integerList.remove(param);
+        }
+    };
+
+    private class RemoveAllElementTask implements Task<Integer, List<Integer>> {
+        @Override
+        public void execute(Integer param, List<Integer> integerList) {
+            integerList.removeAll(Arrays.asList(param, param + 1));
+        }
+    };
+
+    private class RetainAllElementTask implements Task<Integer, List<Integer>> {
+        @Override
+        public void execute(Integer param, List<Integer> integerList) {
+            integerList.retainAll(Arrays.asList(param, param + 1));
+        }
+    };
+
+    private class SetElementTask implements Task<Integer, List<Integer>> {
+        @Override
+        public void execute(Integer param, List<Integer> integerList) {
+            integerList.set(param,10);
+        }
+    };
+
+    private class SetIteratorElementTask implements Task<Integer, ListIterator<Integer>> {
+        @Override
+        public void execute(Integer param, ListIterator<Integer> i) {
+            for (int j=0; j < param; j++) i.next();
+            i.set(10);
+        }
+    };
+    private class RemoveIteratorElementTask implements Task<Integer, ListIterator<Integer>> {
+        @Override
+        public void execute(Integer param, ListIterator<Integer> i) {
+            for (int j=0; j < param; j++) i.next();
+            i.remove();
+        }
+    };
+    private class AddIteratorElementTask implements Task<Integer, ListIterator<Integer>> {
+        @Override
+        public void execute(Integer param, ListIterator<Integer> i) {
+            for (int j=0; j < param; j++) i.next();
+            i.add(11);
+        }
+    };
+    private class ClearElementsTask implements Task<Integer, List<Integer>> {
+        @Override
+        public void execute(Integer param, List<Integer> i) {
+            i.clear();
+        }
+    };
+
+    private interface Task<P, T> {
+        void execute(P param, T i);
+    }
+}
```

Comment:
Not a proper review @dota17 as this would require more time than I have right now. But this file is missing the extra newline for git. :+1: and thanks for all your contributions :tada: